### PR TITLE
add apex test

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

follow-up for #2865

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/models/test_amp.py [hunk 1]
```diff
@@ -173,3 +173,15 @@ def test_cpu_model_with_amp(tmpdir):
 
     with pytest.raises((MisconfigurationException, ModuleNotFoundError)):
         tpipes.run_model_test(trainer_options, model, on_gpu=False)
+
+
+def test_amp_with_apex():
```
Comment (williamFalcon) [inline tests/models/test_amp.py:178]: apex still needs the choice of O1, O2, etc... how is that being done right now? we used to have a flag for this.

also, call it:
amp_backend='native | apex'

**Comment by Borda** [comment]
the failing test is unrelated and shall be fixed in #2944

File: tests/models/test_amp.py [hunk 2]
```diff
@@ -173,3 +174,38 @@ def test_cpu_model_with_amp(tmpdir):
...
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        precision=16,
+        amp_backend='apex',
+        gpus=1,
+    )
+    assert str(trainer.amp_backend) == "AMPType.APEX"
+    trainer.fit(model)
+    assert trainer.state == TrainerState.FINISHED
+    assert trainer.dev_debugger.count_events('AMP') >= 1
```
Comment (williamFalcon) [inline tests/models/test_amp.py:211]: assert the exact number of scaling ops that is expected... this is good error checking in general.

File: tests/models/test_amp.py [hunk 3]
```diff
@@ -173,3 +174,38 @@ def test_cpu_model_with_amp(tmpdir):
 
     with pytest.raises((MisconfigurationException, ModuleNotFoundError)):
         tpipes.run_model_test(trainer_options, model, on_gpu=False)
+
+
+def test_amp_without_apex(tmpdir):
```
Comment (williamFalcon) [inline tests/models/test_amp.py:179]: i don't understand what this is testing

Comment (Borda) [reply tests/models/test_amp.py:179]: this is testing that even with default amp type without requesting percsion=16 the amp backend is void...

**Pull request merged**